File: ruby/functions/function_calling_function.rb
# Copyright (C) 2019-2026 Katarzyna Krasińska
# Languagium - https://github.com/katheroine/languagium
# Licensed under GPL-3.0 - see LICENSE.md

def inside()
  puts("* Inside.")
  return "IN"
end

def outside()
  puts("# Outside:\n",
    "# Calling function from function...")
  result = inside()
  puts("# result: #{result}\n")
  return "OUT"
end

puts("Calling function...")
result = outside()
puts("result: #{result}\n")

File: c++/functions/function_calling_function.cpp
/*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 */
// SPDX-License-Identifier: GPL-3.0-or-later

#include <iostream>

const char* inside()
{
  std::cout << "* Inside.\n";
  return "IN";
}

const char* outside()
{
  std::cout << "# Outside:\n"
    "# Calling function from function...\n";
  const char* result = inside();
  std::cout << "# result: " << result << std::endl;
  return "OUT";
}

int main()
{
  std::cout << "Calling function...\n";
  const char* result = outside();
  std::cout << "result: " << result << std::endl;
}
